apps/x509.c

```diff
@@ -943,7 +943,7 @@ int x509_main(int argc, char **argv)
         } else if (i == modulus) {
             BIO_printf(out, "Modulus=");
             if (EVP_PKEY_is_a(pkey, "RSA")) {
-                BIGNUM *n;
+                BIGNUM *n = NULL;
```

Comment:
There is a very similar thing below. Should that be fixed as well? Do we properly document that the bignums passed to `EVP_PKEY_get_bn_param()` must be properly initialized?

Reply:
And should we check all EVP_PKEY_get_bn_param() calls in ouir code whether this invariant is correct?

Reply:
Docs say:

> EVP_PKEY_get_bn_param() retrieves a key I<pkey> BIGNUM value I<**bn>
associated with a name of I<key_name>. If I<*bn> is NULL then the BIGNUM
is allocated by the method.

This seems sufficient to me.

Reply:
> This seems sufficient to me.

Yep. Then the call below needs to be fixed as well. Not sure if we need to review all calls.

Reply:
Only fifty such calls....

Reply:
There are only two others which are fixed in #15908